### ISSUE TITLE
Fatal Exception: java.lang.OutOfMemoryError with the TelemetryClient

### ISSUE THREAD

**Issue by adavis** [mapbox/mapbox-java#467]
Seeing random crashes in my application.

**Platform:** Android 6.0 and 6.01
**Mapbox Services version:** 5.0.2

### Expected behavior

No crash

### Actual behavior

```
Fatal Exception: java.lang.OutOfMemoryError: Failed to allocate a 662262 byte allocation with 652224 free bytes and 636KB until OOM
       at java.lang.AbstractStringBuilder.enlargeBuffer(AbstractStringBuilder.java:95)
       at java.lang.AbstractStringBuilder.append0(AbstractStringBuilder.java:133)
       at java.lang.StringBuilder.append(StringBuilder.java:124)
       at org.json.JSONStringer.string(JSONStringer.java:344)
       at org.json.JSONStringer.key(JSONStringer.java:375)
       at org.json.JSONObject.writeTo(JSONObject.java:720)
       at org.json.JSONStringer.value(JSONStringer.java:237)
       at org.json.JSONArray.writeTo(JSONArray.java:613)
       at org.json.JSONArray.toString(JSONArray.java:585)
       at com.mapbox.services.android.telemetry.http.TelemetryClient.sendEventsWrapped(TelemetryClient.java:202)
       at com.mapbox.services.android.telemetry.http.TelemetryClient.sendEvents(TelemetryClient.java:117)
       at com.mapbox.services.android.telemetry.MapboxTelemetry.flushEventsQueueImmediately(MapboxTelemetry.java:516)
       at com.mapbox.services.android.telemetry.MapboxTelemetry.putEventOnQueue(MapboxTelemetry.java:402)
       at com.mapbox.services.android.telemetry.MapboxTelemetry.addLocationEvent(MapboxTelemetry.java:444)
       at com.mapbox.services.android.telemetry.TelemetryLocationReceiver.onReceive(TelemetryLocationReceiver.java:27)
       at android.support.v4.content.LocalBroadcastManager.executePendingBroadcasts(LocalBroadcastManager.java:297)
       at android.support.v4.content.LocalBroadcastManager.access$000(LocalBroadcastManager.java:46)
       at android.support.v4.content.LocalBroadcastManager$1.handleMessage(LocalBroadcastManager.java:116)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by zugaldia** [comment]
Thanks for the report @adavis, I'll look into this. I believe we tackled this OOM error with https://github.com/mapbox/mapbox-java/pull/390 but I need to double check that the fix was in the release mentioned above.

**Comment by zugaldia** [comment]
@adavis Looks like the fix for this issue didn't make it to the Maps SDK until version 5.1.0 (https://github.com/mapbox/mapbox-gl-native/commit/a96c6ee82e8fbb824e68fdec6d699e0a1a5bbfde) which is currently on beta. Let's keep the issue open until we have a final release, please let me know if you still the issue once you upgrade.

**Comment by adavis** [comment]
ok, thanks for the update 👍🏽

**Comment by cammace** [comment]
Looks like the fix made it into the final 5.1.0 release. Closing but if the issue still occurs I can reopen.
